docs/design/fsdp-design-en.md

```diff
@@ -0,0 +1,373 @@
+# DLRover's FSDP Elastic Training Design Document
+
+## 1. Background
+
+As the scale of machine learning models has grown, the necessity for distributed
+training has become paramount. DLRover is especially proficient in addressing
+challenges such as node failures, dynamic resource allocation, and automatic
+recovery. While PyTorch provides support through FSDP (Fully Sharded Data Parallel),
+DLRover still faces specific issues related to dynamic resources and fault tolerance.
+These problems often manifest as OOM (Out of Memory) errors and complications in
+reloading sharded checkpoints when the world size alters."
+
+### 1.1 Motivation
+
+DLRover aims to support FSDP at the system level for:
+
+1. **System-Level FSDP Elastic Training:** DLRover should recognize FSDP
+training, aiding users in auto-saving and loading checkpoints.
+2. **Enhanced PyTorch FSDP Resharding:** PyTorch's save method requires
+refinement for dynamic resharding when world size varies.
+
```

Comment:
PyTorch's save method requires
refinement for dynamic resharding when world size varies.

save method means?

Reply:
world size? to word?

Reply:
Sorry for the confusion, my wording was imprecise. I'll make corrections. What I meant to say was: in order to save the model and optimizer state, we utilize the save and load functions. However, the current torch.save and torch.load in PyTorch don't support save and load in a resharding mode, which would be preferable when the world size changes.